### DISCUSSION TITLE
Shelly Plus 2PM doesn't post changes via MQTT

### DISCUSSION THREAD

**Discussion by Meinhard-Risch** [arendst/Tasmota#19731] [Q&A]
I've flashed a Shelly Plus 2PM with the tasmota32solo1.bin. Everything works fine. Over the Web it is possible to toggle the relais. Switchmode 3 is set. The connection via mqtt is stable. I'm able to set the POWER 1 and the POWER2 to false or true to switch the relais ON or OFF. But the result is never talked back to the MQTT-Server.  The Tasmota-Device doesn't publish any changes.  The telemetry period is set to default (300).
I hope, someone could help.

**Comment by barbudor** [comment]
Which mqtt broker? Mosquitto? Version?

You may have ACL enabled

**Comment by Meinhard-Risch** [comment]
Its  MQTT via SONOFF-Adapter in ioBroker. Sorry, but what does it mean: ACL?

**Comment by Meinhard-Risch** [comment]
Oh sorry, now i'm clear: No, there is no Acces-Control-List.

**Reply by barbudor** [reply]
I don't know ioBroker but if Tasmota is connected, can receive commands from MQTT but is not successfull in publishing, I would bet it is problem with ioBroker configuration.

**Reply by sfromis** [reply]
Do make sure that Tasmota has a working MQTT connection. When powering on, I see such in the Tasmota console:
```
22:53:18.530 MQT: stat/devicetopic/POWER = {"POWER":"ON"}
22:53:18.532 MQT: stat/devicetopic/POWER = ON
```
If you get `RSL:` instead of `MQT:`, Tasmota did not reach the broker.

**Comment by Meinhard-Risch** [comment]
Thank you barbudor for helping. I just restarted the device and got the following in the console:
_22:05:10.659 MQT: Connected
22:05:10.665 MQT: tele/ShellyTasmota08/LWT = Online (retained)
22:05:10.669 MQT: cmnd/ShellyTasmota08/POWER = 
22:05:10.670 MQT: Subscribe to cmnd/ShellyTasmota08/#
22:05:10.671 MQT: Subscribe to cmnd/tasmotas/#
22:05:10.683 MQT: Subscribe to cmnd/ShellyTasmota08_fb/#
22:05:10.686 MQT: tele/ShellyTasmota08/INFO1 = {"Info1":{"Module":"Shelly Plus 2PM PCB v0.1.9","Version":"13.1.0(solo1)single-core","FallbackTopic":"cmnd/ShellyTasmota08_fb/","GroupTopic":"cmnd/tasmotas/"}}
22:05:10.707 WIF: IPv6 Local fe80::ab6:1fff:fecc:8e68
22:05:10.723 MQT: tele/ShellyTasmota08/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"ShellyTasmota08","IPAddress":"192.168.101.95","IP6Global":"","IP6Local":"fe80::ab6:1fff:fecc:8e68"}}
22:05:10.741 MQT: tele/ShellyTasmota08/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":9}}
22:05:10.748 MQT: stat/ShellyTasmota08/RESULT = {"POWER1":"OFF"}
22:05:10.750 MQT: stat/ShellyTasmota08/POWER1 = OFF
22:05:10.763 MQT: stat/ShellyTasmota08/RESULT = {"POWER2":"OFF"}
22:05:10.764 MQT: stat/ShellyTasmota08/POWER2 = OFF
22:05:10.768 TFS: File 'autoexec.bat' not found
22:05:14.003 QPC: Reset
22:05:14.773 WIF: IPv6 Global 2001:9e8:c0e2:7c00:ab6:1fff:fecc:8e68
22:05:15.025 MQT: tele/ShellyTasmota08/STATE = {"Time":"2023-10-13T22:05:15","Uptime":"0T00:00:09","UptimeSec":9,"Heap":129,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":24,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":46},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"SicheresNetz123","BSSId":"E8:DF:70:18:2F:44","Channel":1,"Mode":"11n","RSSI":40,"Signal":-80,"LinkCount":1,"Downtime":"0T00:00:03"}}
22:05:15.072 MQT: tele/ShellyTasmota08/SENSOR = {"Time":"2023-10-13T22:05:15","Switch1":"OFF","Switch2":"OFF","ANALOG":{"Temperature1":52.0},"ENERGY":{"TotalStartTime":"2023-10-13T18:51:13","Total":0.000,"Yesterday":0.000,"Today":0.000,"Period":[0,0],"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":0,"Current":[0.000,0.000]},"TempUnit":"C"}
22:05:15.119 NRG: Energy reset by invalid data
22:05:15.234 HTP: Main Menu
22:05:15.998 APP: Boot Count 9
22:05:16.344 CFG: Saved, Count 111, Bytes 4096
22:05:17.335 HTP: Consoles
22:05:19.140 HTP: Console_

I Think, it looks normal ...

**Reply by sfromis** [reply]
Then Tasmota **is** doing the publish operations. If you do not see that in ioBroker, make sure that you do the subscribes right, and if the MQTT setup is "connected right".

**Comment by Meinhard-Risch** [comment]
Hi barbudor, you've won (_I would bet it is problem with ioBroker configuration.)_!
I just wrote a little javascript to attach another light to the endpoint. And what should I say. The stat is written at the endpiont of MQTT-Server, but the value is not shown by the sonoff-adapter. I'm sorry to waste your time. Thank you!
